pkg/eks/cache.go

```diff
@@ -0,0 +1,34 @@
+package eks
+
+import (
+	"time"
+
+	awseks "github.com/aws/aws-sdk-go/service/eks"
+)
+
+const clusterInfoCacheTTL = 15 * time.Second
+
+type clusterInfo struct {
+	timestamp time.Time
+	cluster   *awseks.Cluster
+}
+
+func (c *ClusterProvider) clusterInfoNeedsUpdate() bool {
+	if c.Status.clusterInfo == nil {
+		return true
+	}
+	if c.Status.clusterInfo == nil {
+		return true
+	}
+	if time.Since(c.Status.clusterInfo.timestamp) > clusterInfoCacheTTL {
+		return true
+	}
+	return false
+}
+
+func (c *ClusterProvider) newClusterInfo(cluster *awseks.Cluster) {
```

Comment:
To follow conventions, I think this should be renamed to `setClusterInfo(cluster *awseks.Cluster)`